web-registry/src/lib/normalizers/normalizeProjectsWithOrderData.ts

```diff
@@ -0,0 +1,55 @@
+import { ProjectInfo } from '@regen-network/api/lib/generated/regen/ecocredit/v1/query';
+
+import { GECKO_PRICES } from 'pages/Projects/hooks/useProjectsSellOrders.types';
+import {
+  getPurchaseInfo,
+  normalizeToUISellOrderInfo,
+} from 'pages/Projects/hooks/useProjectsSellOrders.utils';
+import { ProjectWithOrderData } from 'pages/Projects/Projects.types';
+import { SellOrderInfoExtented } from 'hooks/useQuerySellOrders';
+
+import DefaultProject from 'assets/default-project.jpg';
+
+interface NormalizeProjectsWithOrderDataParams {
+  projects?: ProjectInfo[];
+  sellOrders?: SellOrderInfoExtented[];
+  geckoPrices?: GECKO_PRICES;
+  userAddress?: string;
+  metadatas?: any[];
+}
+
+export const normalizeProjectsWithOrderData = ({
```

Comment:
This is an existing function that has been extracted from its hook.
Also, metadata fetches there were previously done here have been moved elsewhere to keep the function without async code.